.github/workflows/docker.yml

```diff
@@ -34,32 +34,22 @@ jobs:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
 
-      # TODO: consider using https://github.com/docker/metadata-action instead of this
-      # custom magic
       - name: Calculate docker image tag
         id: set-tag
-        run: |
-          case "${GITHUB_REF}" in
-              refs/heads/develop)
-                  tag=develop
-                  ;;
-              refs/heads/master|refs/heads/main)
-                  tag=latest
-                  ;;
-              refs/tags/*)
-                  tag=${GITHUB_REF#refs/tags/}
-                  ;;
-              *)
-                  tag=${GITHUB_SHA}
-                  ;;
-          esac
-          echo "::set-output name=tag::$tag"
+        uses: docker/metadata-action@master
+        with:
+          images: matrixdotorg/synapse
+          tags: |
+            type=ref,event=branch
```

Comment:
Yes, this is indeed a problem, for the time being I haven't come up with any solutions. Do you have any good idea on how to solve this?

Reply:
Hm, the only thing I can think of right now is to have three lines like so:
```
type=raw,value=develop,enable=${{ github.ref == "refs/heads/develop" }}
type=raw,value=latest,enable=${{ github.ref == "refs/heads/master" }}
type=raw,value=latest,enable=${{ github.ref == "refs/heads/main" }}
```
corresponding to the three cases in the previous shell script. It's not pretty and I'm open to better solutions.

Reply:
Thank you very much. Yes this would work, although not very elegant.

Hi @DMRobertson , do you have any comment or suggestion on this?

Or maybe we should go with this solution?

Reply:
I'm no authority here, but I think @squahtx's suggestion makes sense. I think the most important outstanding task is to test it in a fork, as Sean explains [here](https://github.com/matrix-org/synapse/pull/12573#pullrequestreview-957964692).

Reply:
Try to resolve this with ce281687, please review

Reply:
Looks good!